src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -886,28 +885,14 @@ private static boolean isAstInside(DetailAST tree, DetailAST ast) {
      * @param ident ident token.
      * @return true if field can be referenced from a static context.
      */
-    private boolean canBeReferencedFromStaticContext(DetailAST ident) {
-        AbstractFrame variableDeclarationFrame = findFrame(ident, false);
-        while (variableDeclarationFrame.getType() == FrameType.BLOCK_FRAME
```

Comment:
This mutation had a cascading effect, and so the end result is the frame of the `ident` does not matter.

According to the callers, the frame can only be `CLASS_FRAME`, `METHOD_FRAME`, `BLOCK_FRAME`. So the question is why does it not matter if act on a `BLOCK_FRAME`.

This requires more in-depth analysis.

Reply:
This is cascaded from https://github.com/checkstyle/checkstyle/pull/12628#discussion_r1068792628 and depends on it's analysis.